src/translate/function.rs

```diff
@@ -254,6 +254,23 @@ pub fn translate_function(sql_function: &SqlFunction) -> Result<Expr> {
             })))
         }
         "REVERSE" => func_with_one_arg!(Function::Reverse),
+        "SUBSTR" => {
+            check_len_range(name, args.len(), 2, 3)?;
+
+            let expr = translate_expr(args[0])?;
+            let start = translate_expr(args[1])?;
+            let count = if args.len() == 2 {
+                None
+            } else {
+                Some(translate_expr(args[2])?)
+            };
```

Comment:
fyi. Let's get familiar with using methods provided by `bool` and `Option` :)
https://doc.rust-lang.org/std/primitive.bool.html#method.then
https://doc.rust-lang.org/std/option/enum.Option.html#method.transpose

```rust
let count = (args.len() > 2)         
    .then(|| translate_expr(args[2]))
    .transpose()?;
```

Reply:
Thanks for the information.
Edited as you said.